og.module

```diff
@@ -25,22 +25,29 @@ define('OG_DEFAULT_ACCESS_FIELD', 'og_roles_permissions');
 
 /**
  * Implements hook_entity_insert().
+ *
+ * Subscribe the group manager.
  */
 function og_entity_insert(EntityInterface $entity) {
   if (!Og::isGroup($entity->getEntityTypeId(), $entity->bundle())) {
+    // Not a group entity.
+    return;
+  }
+
+  if (!$entity instanceof EntityOwnerInterface) {
     return;
   }
 
-  if ($entity instanceof EntityOwnerInterface) {
-    // Subscribe the group manager.
-    $membership = OgMembership::create(['type' => OgMembershipInterface::TYPE_DEFAULT]);
-    $membership
-      ->setUser($entity->getOwnerId())
-      ->setEntityId($entity->id())
-      ->setGroupEntityType($entity->getEntityTypeId())
-      ->setFieldName($membership->getFieldName())
-      ->save();
+  if (!$entity->getOwner()) {
+    // User is anonymous, se we cannot set a membership for them.
+    return;
   }
+
+  $membership = OgMembership::create(['type' => OgMembershipInterface::TYPE_DEFAULT]);
+  $membership
+    ->setUser($entity->getOwner())
+    ->setGroup($entity)
+    ->save();
```

Comment:
Just a thought now that we are improving DX in this PR: every time we create an `OgMembership` we are setting the type like this. Wouldn't it be a great idea to set the default value for the type in `OgMembership::baseFieldDefinitions()`, so that we don't need to specify it all the time?

We can still override it if we need to have a different type.